client/examples/repl/command.rs

```diff
@@ -0,0 +1,384 @@
+
+        let mut client_path = r.borrow_mut();
+        client_path.clear();
+        client_path.extend_from_slice(parameters[0].as_bytes());
+
+        stronghold
+            .as_ref()
+            .unwrap()
+            .create_client(client_path.deref().clone())?;
+
+        Ok(Default::default())
+    }
+
+    fn name(&self) -> String {
+        "init".to_string()
+    }
+
+    fn required_parameters(&self) -> Vec<String> {
+        vec!["<client_path>"].iter().map(|s| s.to_string()).collect()
+    }
+}
+
+#[derive(Default)]
+pub struct GenerateKeyCommand;
+impl Command for GenerateKeyCommand {
+    fn eval(&self, context: State, parameters: &[String]) -> Result<TermAction, ReplError<String>> {
+        let stronghold = context.stronghold.borrow().clone().unwrap();
+        let client_path = context.client_path.borrow();
+
+        let kt = parse_keytype(&parameters[0])?;
+        let vp = &parameters[1];
+        let rp = &parameters[2];
+        let client = stronghold.get_client(client_path.deref())?;
+
+        // execute the procedure
+        client.execute_procedure(GenerateKey {
+            ty: kt,
+            output: Location::const_generic(vp.clone().into_bytes(), rp.clone().into_bytes()),
+        })?;
+
+        Ok(TermAction::OkMessage("Key stored sucessfully".to_string()))
+    }
+
+    fn name(&self) -> String {
+        "keygen".to_string()
+    }
+
+    fn required_parameters(&self) -> Vec<String> {
+        vec!["<key_type>", "<vault_path>", "<record_path>"]
+            .iter()
+            .map(|s| s.to_string())
+            .collect()
+    }
+}
+
+#[derive(Default)]
+pub struct CheckVaultCommand;
+impl Command for CheckVaultCommand {
+    fn eval(&self, context: State, parameters: &[String]) -> Result<TermAction, ReplError<String>> {
+        let stronghold = context.stronghold.borrow().clone().unwrap();
+        let client_path = context.client_path.borrow();
+
+        let client = stronghold.get_client(client_path.deref())?;
+        let vp = &parameters[0];
+
+        match client.vault_exists(vp) {
+            Ok(exists) => {
+                println!("Vault exists? {}", if exists { "yes" } else { "no" });
+            }
+            Err(e) => return Err(e.into()),
+        }
+
+        Ok(Default::default())
+    }
+
+    fn name(&self) -> String {
+        "checkvault".to_string()
+    }
+
+    fn required_parameters(&self) -> Vec<String> {
+        vec!["<vault_path>"].iter().map(|s| s.to_string()).collect()
+    }
+}
+
+#[derive(Default)]
+pub struct CheckRecordCommand;
+impl Command for CheckRecordCommand {
+    fn eval(&self, context: State, parameters: &[String]) -> Result<TermAction, ReplError<String>> {
+        let stronghold = context.stronghold.borrow().clone().unwrap();
+        let client_path = context.client_path.borrow();
+        let client = stronghold.get_client(client_path.deref())?;
+
+        let vp = &parameters[1];
+        let rp = &parameters[2];
```

Comment:
This currently panics.

```suggestion
        let vp = &parameters[0];
        let rp = &parameters[1];
```